tests/example_app_with_oauth/src/app.py

```diff
@@ -0,0 +1,185 @@
+import httpx
+
+from soar_sdk.abstract import SOARClient
+from soar_sdk.action_results import ActionOutput, OutputField
+from soar_sdk.app import App
+from soar_sdk.asset import AssetField, BaseAsset
+from soar_sdk.auth import (
+    AuthorizationCodeFlow,
+    ClientCredentialsFlow,
+    OAuthBearerAuth,
+    OAuthConfig,
+    SOARAssetOAuthClient,
+)
+from soar_sdk.logging import getLogger
+from soar_sdk.params import Params
+from soar_sdk.webhooks.models import WebhookRequest, WebhookResponse
+
+logger = getLogger()
+
+APP_NAME = "example_app_with_oauth"
+APP_ID = "9b388c08-67de-4ca4-817f-26f8fb7cbf56"
+BASE_URL = "https://graph.microsoft.com"
+DEFAULT_SCOPE = "https://graph.microsoft.com/.default"
+
+
+class Asset(BaseAsset):
+    auth_type: str = AssetField(
+        default="credentials",
+        value_list=["credentials", "interactive"],
+        description="Authentication method",
+    )
+    client_id: str = AssetField(description="OAuth Client ID")
+    client_secret: str = AssetField(
+        sensitive=True,
+        description="OAuth Client Secret",
+    )
+    tenant_id: str = AssetField(description="Azure AD Tenant ID")
+
+    @property
+    def token_url(self) -> str:
+        return f"https://login.microsoftonline.com/{self.tenant_id}/oauth2/v2.0/token"
+
+    @property
+    def auth_url(self) -> str:
+        return (
+            f"https://login.microsoftonline.com/{self.tenant_id}/oauth2/v2.0/authorize"
+        )
```

Comment:
Probably generalize this a bit more. We can make it MS-flavored for integration tests if need be:


```suggestion
class Asset(BaseAsset):
    domain: str = AssetField(
      required=True,
      description="URL of service to authenticate against, e.g. https://login.notarealwebsite.com",
    ),
    auth_type: str = AssetField(
        default="credentials",
        value_list=["credentials", "interactive"],
        description="Authentication method",
    )
    client_id: str = AssetField(description="OAuth Client ID")
    client_secret: str = AssetField(
        sensitive=True,
        description="OAuth Client Secret",
    )
    tenant_id: str = AssetField(description="Service Tenant ID. Appended to domain")
    token_uri: str = AssetField(default="/oauth2/v2.0/token")
    auth_uri: str = AssetField(default="/oauth2/v2.0/authorize")


    @property
    def token_url(self) -> str:
        return f"{self.domain.removesuffix('/')}/{self.tenant_id}/{self.token_uri.removeprefix('/')}"

    @property
    def auth_url(self) -> str:
        return (
          f"{self.domain.removesuffix('/')}/{self.tenant_id}/{self.auth_uri.removeprefix('/')}"
        )
```